test/test_cooperative_wait.jl

```diff
@@ -1,7 +1,7 @@
 # tests for the various kinds of waits
 include("common.jl")
 
-MPI.Init()
+MPI.Init(threadlevel=:multiple)
```

Comment:
Isn't that our default?

Reply:
Nope, the default is ``:serialized`, see https://github.com/JuliaParallel/MPI.jl/blob/a57528f7686431a47e009faf882d16ec23eacaeb/src/environment.jl#L114

Reply:
Is this always the case here or only if one launches Julia with `threads > 1` ?

Reply:
Good question.  I guess it would be only when julia is launched with more than one thread.  But Valentin would know better.

Reply:
We went with `:serialized` as that is the most common use case, and not all MPI libraries support `:multiple` by default (or they might incur additional overhead)

Reply:
So ideally one should always have 
```
(Threads.nthreads() > 1) ? MPI.Init(threadlevel=:multiple) : MPI.Init()
```

Reply:
This doesn't account for interactive threadpool usage, or foreign threads being dynamically added.

`maxthreadid` is probably okay, but that get's skewed by GC threads and the IO thread.